### DISCUSSION TITLE
I am using web3modal version 1.9.9

### DISCUSSION THREAD

**Discussion by degenSumi** [WalletConnect/walletconnect-monorepo#2829] [Developer Support] [answered]
Hey please help us save our dapp we are using web3modal version 1.9.9 to connect and sign this is our connection logic
`import { useEffect, useState, createContext, ReactNode } from 'react';
import Web3Modal from 'web3modal';
import { ethers } from 'ethers';
import { useContextState } from '@/context';
import CoinbaseWalletSDK from "@coinbase/wallet-sdk";
import WalletConnect from "@walletconnect/web3-provider";

const web3modalStorageKey = 'WEB3_CONNECT_CACHED_PROVIDER';

export const WalletContext = createContext<any>({});

const providerOptions = {
  // walletlink: {
  //   package: CoinbaseWalletSDK, // Required
  //   options: {
  //     appName: "Web3Modal Demo", // Required
  //     infuraId: '65926722c54b421ba5eff41537943d62', // Required unless you provide a JSON RPC url; see `rpc` below
  //   },
  // },
  walletconnect: {
    package: WalletConnect, // required
    options: {
      infuraId: '65926722c54b421ba5eff41537943d62', // required
    },
  },
};

export const WalletProvider = ({ children }: { children: ReactNode }) => {
  const [address, setAddress] = useState<string | undefined>(undefined);
  const [balance, setBalance] = useState<string | undefined>(undefined);
  const [loading, setLoading] = useState<boolean>(false);
  const [error, setError] = useState<boolean>(false);
  const [{ refreshBalance },]: any = useContextState();
  const [provider, setProvider] = useState();
  const [library, setLibrary] = useState();
  const [account, setAccount] = useState();

  const web3Modal =
    typeof window !== 'undefined' && new Web3Modal({ cacheProvider: true, providerOptions});

  /* This effect will fetch wallet address if user has already connected his/her wallet */
  // useEffect(() => {
  //   async function checkConnection() {
  //     try {
  //       if (window && window.ethereum) {
  //         // Check if web3modal wallet connection is available on storage
  //         if (localStorage.getItem(web3modalStorageKey)) {
  //           await connectToWallet();
  //         }
  //       } else {
  //         console.log('window or window.ethereum is not available');
  //       }
  //     } catch (error) {
  //       console.log(error, 'Catch error Account is not connected');
  //     }
  //   }
  //   checkConnection();
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, []);

  useEffect(() => {
    if (web3Modal.cachedProvider) {
      connectToWallet();
    }
  }, []);

  useEffect(() => {
    if (refreshBalance)
      try {
        (async () => {
          // checkIfExtensionIsAvailable();
          const provider = web3Modal && (await web3Modal.connect());
          const library = new ethers.providers.Web3Provider(provider);
          const signer = library.getSigner();
          if (signer) {
            const web3Address = await signer.getAddress();
            getBalance(library, web3Address)
          }
        })()
        setLoading(true);

      } catch (error) {
        console.log(
          error,
          'got this error on connectToWallet catch block while connecting the wallet'
        );
      }
  }, [refreshBalance])

  const setWalletAddress = async (provider: any) => {
    try {
      const signer = provider.getSigner();
      if (signer) {
        const web3Address = await signer.getAddress();
        setAddress(web3Address);
        getBalance(provider, web3Address);
      }
    } catch (error) {
      console.log(
        'Account not connected; logged from setWalletAddress function'
      );
    }
  };

  const getBalance = async (provider: any, walletAddress: string) => {
    const walletBalance = await provider.getBalance(walletAddress);
    const balanceInEth = ethers.utils.formatEther(walletBalance);
    setBalance(balanceInEth);
  };

  const disconnectWallet = () => {
    setAddress(undefined);
    web3Modal && web3Modal.clearCachedProvider();
  };

  const checkIfExtensionIsAvailable = () => {
    if (
      (window && window.web3 === undefined) ||
      (window && window.ethereum === undefined)
    ) {
      setError(true);
      web3Modal && web3Modal.toggleModal();
    }
  };

  const connectToWallet = async () => {
    try {
      setLoading(true);
      // checkIfExtensionIsAvailable();
      const provider = web3Modal && (await web3Modal.connect());
      const library = new ethers.providers.Web3Provider(provider);
      await subscribeProvider(provider);
      setLibrary(library);
      setProvider(provider);
      setWalletAddress(library);
      setLoading(false);
    } catch (error) {
      setLoading(false);
      console.log(
        error,
        'got this error on connectToWallet catch block while connecting the wallet'
      );
    }
  };

  const subscribeProvider = async (connection: any) => {
    connection.on('close', () => {
      disconnectWallet();
    });
    connection.on('accountsChanged', async (accounts: string[]) => {
      if (accounts?.length) {
        setAddress(accounts[0]);
        const provider = new ethers.providers.Web3Provider(connection);
        getBalance(provider, accounts[0]);
      } else {
        disconnectWallet();
      }
    });
  };

  return (
    <WalletContext.Provider
      value={{
        address,
        balance,
        loading,
        library,
        provider,
        error,
        connectToWallet,
        disconnectWallet,
      }}
    >
      {children}
    </WalletContext.Provider>
  );
};`

**Comment by chadyj** [comment, marked as answer]
Hey @SumitCMLab. 

Web3modal version 1 is deprecated. Here is our guide on how to upgrade https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#web3modal-v10
